financialaid/views_test.py

```diff
@@ -67,39 +74,61 @@ def setUp(self):
+        [49999, "USD", 50000, False],
+        [50000, "USD", 50000, False],
+        [50001, "USD", 50000, True],
+        # Test with an exchange rate greater than 1
+        [49999 * ABC_EXCHANGE_RATE, "ABC", 50000, False],
+        [50000 * ABC_EXCHANGE_RATE, "ABC", 50000, False],
+        [50001 * ABC_EXCHANGE_RATE, "ABC", 50000, True],
+        # Test with an exchange rate less than 1
+        [49999 * XYZ_EXCHANGE_RATE, "XYZ", 50000, False],
+        [50000 * XYZ_EXCHANGE_RATE, "XYZ", 50000, False],
+        [50001 * XYZ_EXCHANGE_RATE, "XYZ", 50000, True],
+    )
+    @ddt.unpack
+    def test_income_validation(self, original_income, original_currency, income_threshold, auto_approved):
         """
-        Tests FinancialAidRequestView post endpoint for auto-approval
+        Tests FinancialAidRequestView post endpoint
         """
+        CountryIncomeThreshold.objects.filter(country_code=self.profile.country).update(
+            income_threshold=income_threshold
+        )
+        data = {
+            "original_income": original_income,
+            "original_currency": original_currency,
+            "program_id": self.program.id,
+        }
         assert FinancialAid.objects.count() == 0
         assert FinancialAidAudit.objects.count() == 0
-        self.data["original_income"] = self.country_income_threshold_50000.income_threshold+1
-        self.assert_http_status(self.client.post, self.request_url, status.HTTP_201_CREATED, data=self.data)
+        self.assert_http_status(self.client.post, self.request_url, status.HTTP_201_CREATED, data=data)
         assert FinancialAid.objects.count() == 1
         assert FinancialAidAudit.objects.count() == 1
         financial_aid = FinancialAid.objects.first()
-        income_usd = determine_income_usd(self.data["original_income"], self.data["original_currency"])
+        income_usd = determine_income_usd(original_income, original_currency)
         assert financial_aid.tier_program == determine_tier_program(self.program, income_usd)
-        assert financial_aid.status == FinancialAidStatus.AUTO_APPROVED
-        assert financial_aid.income_usd == self.data["original_income"]
+        if not auto_approved:
+            assert financial_aid.status == FinancialAidStatus.PENDING_DOCS
+        else:
+            assert financial_aid.status == FinancialAidStatus.AUTO_APPROVED
+        self.assertAlmostEqual(financial_aid.income_usd, income_usd)
+        assert financial_aid.user == self.profile.user
+        self.assertAlmostEqual(financial_aid.original_income, original_income)
+        assert financial_aid.original_currency == original_currency
+        assert financial_aid.country_of_income == self.profile.country
+        assert financial_aid.country_of_residence == self.profile.country
+        now = datetime.datetime.now(tz=pytz.UTC)
+        five_seconds = datetime.timedelta(0, 5)
+        assert now - five_seconds < financial_aid.date_exchange_rate < now + five_seconds
```

Comment:
i would implement the 3 lines above like so:

```python
now = datetime.datetime.now(tz=pytz.UTC)
date_diff = financial_aid.date_exchange_rate - now
assert abs(date_diff.total_seconds()) < 5
```

that's a stylistic suggestion. you can take or leave it. i would also prefer creating a utility function for this, but it looks like this is the only time we this kind of functionality